src/main/java/org/betonquest/betonquest/compatibility/luckperms/permission/LuckPermsNodeBuilder.java

```diff
@@ -0,0 +1,90 @@
+package org.betonquest.betonquest.compatibility.luckperms.permission;
+
+import net.luckperms.api.context.MutableContextSet;
+import net.luckperms.api.node.types.PermissionNode;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Builder class for LuckPerms {@link PermissionNode}s.
+ */
+public class LuckPermsNodeBuilder {
+
+    /**
+     * Default constructor.
+     */
+    public LuckPermsNodeBuilder() {
+        // Empty
+    }
+
+    /**
+     * Builds a list of {@link PermissionNode}s from the given {@link PermissionData}.
+     *
+     * @param data The {@link PermissionData} to build the {@link PermissionNode}s from.
+     * @return A list of {@link PermissionNode}s.
+     */
+    public List<PermissionNode> getNodes(final PermissionData data) {
+        final List<PermissionNode> buildNodes = new ArrayList<>();
+        for (final String permission : data.permissions) {
+            PermissionNode.Builder builder = nodeBuilder(permission);
+            if (!data.value.isEmpty()) {
```

Comment:
If you changed value to nullable, this also needs to be a null check, may you need to keep the empty check